partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -439,44 +441,21 @@ internal object RexConverter {
             return rex(type, call)
         }
 
-        // coalesce(expr1, expr2, ... exprN) ->
-        //   CASE
-        //     WHEN expr1 IS NOT NULL THEN EXPR1
-        //     ...
-        //     WHEN exprn is NOT NULL THEN exprn
-        //     ELSE NULL END
```

Comment:
Prior modeling of `COALESCE` was slightly incorrect here. Previously, there was an explicit `ELSE NULL` added here, which would always add `NULL` to the return type. The correct equivalence would actually be:
```
COALESCE(v1, v2,..., vN)
==
CASE
    WHEN v1 IS NOT NULL THEN v1
    WHEN v2 IS NOT NULL THEN v2
    ... -- similarly for v3..vN-1
    ELSE vN
END
```

Reply:
Nice catch